content/ADR-203-social-ws-login.md

```diff
@@ -0,0 +1,287 @@
+- An implementation for dcl-crypto in Rust needs to be done to support this use case.
+
+## Conclusion
+
+Explored solutions are:
+- 1: Authenticate the WebSocket Connection
+- 2.a: Use a login message
+- 2.b: the login message is sent by the Server
+- 2.c: include the token on each message
+- 2.d: Hybrid Model
+- 3: The Client logins against the service, and the Synapse Token is not sent to the final user
+
+The primary disadvantages associated with the solution using cookies (`1: Authenticate the WebSocket Connection`) are due to native libraries that do not ensure header management when initiating a WebSocket connection. While there are different solutions for current clients that work, there is not guarantee the flexibility of this solution for future clients. Furthermore, the benefits of this solution are not significant enough to justify the effort required for implementation. The community typically opts for a solution more similar to the second group when authentication is required in a WebSocket.
+
+One of the most straightforward solutions to replace in the future is the `2.c: include the token on each message` solution. Since that solution is minimally intrusive, the implementation cost is low, and there is no cost associated with its removal.
+
+Ideally, the solution should follow an upgrade format initiated by the client, whereby the server asks the client the credentials like AuthChain or the Authentication Token. If the message requesting the token is not received, the service terminates the communication, similar to the HTTP or TLS handshake. Solutions within this category include:
+- `2.a: Use a login message`
+- `2.b: the login message is sent by the Server`
+- `3: The Client logins against the service`
+
+Among those solutions, the benefit of implementing the third one is that the Synapse Token is opaque to the client, making it something internal that can be removed in the future if Synapse dependency is removed. Anyway, the cost of implementing that solution today implies many changes in new dcl libraries like `rpc-rust` and `decentraland-crypto-rust` which may imply a risk for a project as big as Social Service M2 is.
+
+For the Scope of the Social Service M2 Project, the decition is to implement the `2.c` solution, which will handle the authentication individually on each message, as it is the cheaper the implement and the focus and effort is in the robustness and availability of a new Websocket Server implemented in Rust that is thought to replace the dependency to Synapse when managing friendships and request friendships. The `2.c` solution does not represent any security risk or concern, as the Synapse Token is already stored on the local storage of the client, and as it will be validated on every message the behaviour will be analogue on what exists today.
```

Comment:
```suggestion
For the Scope of the Social Service M2 Project, the decision is to implement the `2.c` solution, which will handle the authentication individually on each message, as it is cheaper the implement and the focus and effort is in the robustness and availability of a new WebSocket Server implemented in Rust that is thought to replace the dependency to Synapse when managing friendships and request friendships. The `2.c` solution does not represent any security risk or concern, as the Synapse Token is already stored on the local storage of the client, and as it will be validated on every message the behavior will be analog to what exists today.
```